src/libappimage/desktop_integration/integrator/ResourcesExtractor.cpp

```diff
@@ -0,0 +1,91 @@
+// system
+#include <map>
+
+// local
+#include <appimage/core/PayloadEntryType.h>
+#include "ResourcesExtractor.h"
+
+namespace appimage {
+    namespace desktop_integration {
+        namespace integrator {
+            ResourcesExtractor::ResourcesExtractor(core::AppImage appImage) : appImage(appImage) {}
+
+            DesktopIntegrationResources ResourcesExtractor::extract() {
+                DesktopIntegrationResources resources = {};
+                for (auto fileItr = appImage.files(); fileItr != fileItr.end(); ++fileItr) {
+                    if (fileItr.type() != core::PayloadEntryType::REGULAR && fileItr.type() != core::PayloadEntryType::LINK)
+                        continue;
+
+                    const auto& filePath = *fileItr;
+
+                    if (extractDesktopFile && isMainDesktopFile(filePath)) {
+                        resources.desktopEntryPath = filePath;
```

Comment:
No need to hide them, that's why it's a struct.

Reply:
Every class should define a proper interface, that makes working with them safer and more reliable. E.g., only in a setter you could implement checks for validity of input parameters (so-called preconditions), class invariants or whether the result of the method makes sense (so-called postconditions).

Reply:
The only and sole purpose of this struct is to provide a convenient store for the `ResourceExtractor`. Don't overthink it, it's just a struct.

Reply:
This kind of "ignoring best practices because noone will ever see it" leads to spaghetti code, and is one of the reasons the old code base is in such a bad condition. "Don't overthink it" is a bad excuse.

Reply:
I'm not trying to ignore any best practices here. `DesktopIntegrationResources`  posses no procedures that will affect the data on it and don't perform any kind of validation. Having getters and setters in this scenario is not required at all.

I promise to add all of the getters and setters if any of the conditions exposed above changes.

Reply:
Mind to open issues for "promises"? Then they won't be forgotten.

Reply:
Added a comment on the docstrings. So it will not go away.